lib/src/test/java/io/ably/lib/test/rest/RestInitTest.java

```diff
@@ -235,4 +235,62 @@ public void init_default_log_output_stream() {
 			System.setOut(oldTarget);
 		}
 	}
+
+	/**
+	 * Init library with 'production' environment
+	 * Spec: RSC11
+	 */
+	@Test
+	public void init_production_environment() {
+		try {
+			TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = new ClientOptions(testVars.keys[0].keyStr);
+			opts.environment = "production";
+			AblyRest ably = new AblyRest(opts);
+			assertEquals("Unexpected host mismatch", Defaults.HOST_REST, ably.options.restHost);
+		} catch (AblyException e) {
+			e.printStackTrace();
+			fail("init4: Unexpected exception instantiating library");
+		}
+	}
+
+	/**
+	 * Init library with given environment
+	 * Spec: RSC11
+	 */
+	@Test
+	public void init_given_environment() {
+		final String givenEnvironment = "staging";
+		try {
+			TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = new ClientOptions(testVars.keys[0].keyStr);
+			opts.environment = givenEnvironment;
+			AblyRest ably = new AblyRest(opts);
+			assertEquals("Unexpected host mismatch", String.format("%s-%s", givenEnvironment, Defaults.HOST_REST), ably.options.restHost);
+		} catch (AblyException e) {
+			e.printStackTrace();
+			fail("init4: Unexpected exception instantiating library");
+		}
+	}
+
+	/**
+	 * Init library with given environment and specified host
+	 * Spec: RSC11
+	 */
+	@Test
+	public void init_given_host_environment() {
+		final String givenEnvironment = "staging";
+		final String specifiedHost = "fake.ably.io";
+		try {
+			TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = new ClientOptions(testVars.keys[0].keyStr);
+			opts.restHost = specifiedHost;
+			opts.environment = givenEnvironment;
+			AblyRest ably = new AblyRest(opts);
+			assertEquals("Unexpected host mismatch", specifiedHost, ably.options.restHost);
```

Comment:
This should never be allowed and should raise an exception, not allow this, http://docs.ably.io/client-lib-development-guide/versions/features-0-8/#TO3k2